tests/Aql/EngineInfoContainerCoordinatorTest.cpp

```diff
@@ -272,7 +273,8 @@ TEST(EngineInfoContainerTest,
 
   // Mock the Registry
   fakeit::When(Method(mockRegistry, insert))
-      .Do([&](QueryId id, Query* query, double timeout, bool isPrepared, bool keepLease) {
+    .Do([&](QueryId id, Query* query, double timeout, bool isPrepared, bool keepLease,
+            std::unique_ptr<arangodb::cluster::CallbackGuard>&) {
```

Comment:
I do not get how this works, the guard is not used so it seem to be ok. But shouldn't it go in as rvalue reference?

Reply:
The guard is just there to make sure a callback in the reboot tracker is cleaned up at some point. It's just not used in these tests.
But I agree that it's strange to use an lvalue ref here, while the mocked function gets an rvalue ref. @kvahed

Reply:
fakeit enforces this notation. the callbackguard is not used. we have to get rid of fakeit.